baybe/utils/basic.py

```diff
@@ -16,6 +17,23 @@
 _U = TypeVar("_U")
 
 
+class UnspecifiedType(enum.Enum):
+    """Sentinel indicating an unspecified value when `None` is ambiguous."""
+
+    UNSPECIFIED = "UNSPECIFIED"
+
+    @override
+    def __repr__(self):
+        return "UNSPECIFIED"
+
+    def __bool__(self):
+        raise NotImplementedError(f"'{self!r}' has no Boolean representation.")
+
+
+UNSPECIFIED = UnspecifiedType.UNSPECIFIED
```

Comment:
Yes, of course. When there is `not so obvious stuff`, you can usually expect me to explain the rationale in the corresponding commit, which is also the case here 🙃  Please have a look and let me know if still unclear

Reply:
why is it necessary that flags can also be unspecified (instead of just having a default?)

Reply:
For transparency. When it is possible to explicitly set the flag value in a non-discrete space, it suggests that something should be happening behind the scenes, which is not the case. So forbidding to pass an "invalid" setting makes sense to me, just like other validation steps we have. I've had many discussion with people already about the behavior of these flags because the current logic with defaults.

To me, this is similar to the situation you often have with optimizer flags in botorch/etc where you simply have no clue whether stuff is parsed or just silently dropped (best example: the max steps in the pytorch optimizer, which always required me to use the debugger when I wanted to active it).

What are your thoughts?

Reply:
I think the idea is in general good and reasonable. The way that I see it, this could even be later generalized to other UNSPECIFIED situations by implementing functions similar to the `__bool__` function, right? (Not sure if we will ever need it, but still.)